include/hwmalloc/fancy_ptr/void_ptr.hpp

```diff
@@ -77,7 +77,8 @@ class hw_void_ptr
 #if HWMALLOC_ENABLE_DEVICE
     constexpr VoidPtr device_ptr() const noexcept { return m_data.m_device_ptr; }
 
-    auto device_handle() const noexcept { return m_data.m_device_handle; }
+    auto  device_handle() const noexcept { return m_data.m_device_handle; }
+    auto& device_handle() noexcept { return m_data.m_device_handle; }
```

Comment:
why do we need to be able to change a device handle?

Reply:
A very good question. We have a non const version of the host ptr handle ref and I added that to fix a compilation error. But looking again at my code, I added a const to the use of the handle ref and now I do not need to non const access, so I will remove this. Might also be able to do the same with the host ptr handle ref ...